sim/src/simulator.rs

```diff
@@ -37,7 +37,9 @@ impl Simulator {
     }
 
     pub fn query_storage(&mut self, storage_id: &str) -> &str {
-        self.elements[storage_id].type_specific_info().unwrap()
+        let storage_box = &self.elements[storage_id];
+        let storage = storage_box.as_any().downcast_ref::<Storage>().unwrap();
```

Comment:
Does this cast the storage as a storage object?  If so, could we have an error message if the unwrap fails and the name isn't valid?

Reply:
Technically the purpose of unwrap is just that. It panics if the value is none. I can make this explicit, that is nicer.